src/components/GetMetadata.tsx

```diff
@@ -0,0 +1,187 @@
+import { FC, useState, useCallback } from 'react';
+import { useConnection } from '@solana/wallet-adapter-react';
+import { PublicKey } from '@solana/web3.js';
+import { Metadata } from '@metaplex-foundation/mpl-token-metadata';
+
+export const GetMetadata: FC = () => {
+  const { connection } = useConnection();
+  const [tokenAddress, setTokenAddress] = useState('');
+  const [tokenMetadata, setTokenMetadata] = useState(null);
+  const [loaded, setLoaded] = useState(false);
+  
+
+  const getMetadata = useCallback(async (form) => {
+    const tokenMint = form.tokenAddress;
+    const metadataPDA = await Metadata.getPDA(new PublicKey(tokenMint));
+    const metadata = await Metadata.load(connection, metadataPDA);
+    setTokenMetadata({tokenMetadata, ...metadata.data});
+    setLoaded(true);
+    console.log(tokenMetadata);
+    console.log(metadata.data);
+  }, [tokenAddress])
+
+  return (
+    <>
+    <div className="my-6">
+      <input
+        type="text"
+        value={tokenAddress}
+        className="form-control block mb-2 ml-auto mr-auto max-w-800 px-4 py-2 text-xl font-normal text-gray-700 bg-white bg-clip-padding border border-solid border-gray-300 rounded transition ease-in-out m-0 focus:text-gray-700 focus:bg-white focus:border-blue-600 focus:outline-none"
+        placeholder="Token Address"
+        onChange={(e) => setTokenAddress(e.target.value)}
+      />
+  
+      <button
+        className="px-8 m-2 btn animate-pulse bg-gradient-to-r from-[#9945FF] to-[#14F195] hover:from-pink-500 hover:to-yellow-500 ..."
+        onClick={() => getMetadata({tokenAddress})}>
+          <span>Get Metadata</span>
+      </button>
+    </div>
+    <div className="my-6">
+    {!loaded ? (
```

Comment:
Any way we can make it not load any metadata screen if a mint has not been entered?

Reply:
Okay, that’s what I had originally. I’ll go back to that. I thought it might be nice to show what fields are available so that’s why I added the empty metadata table.